Fixes naming convention and removes empty prometheus metrics and fixes skips

codefromthecrypt opened the pull request:
When we switched to micrometer's prometheus in 2.7.3, we accidentally
broke the naming convention used before.

Notably, in prometheus.yml, we had this:

```yaml
      - source_labels: [__name__]
        regex: '(?:gauge|counter)_zipkin_collector_(.*)_([^_]*)'
        replacement: '${2}'
        target_label: transport

```

As the default naming convention adds "_total", this causes dashboards
to end up with a transport literally named "_total".

This also removes empty metrics eagerly initialized and patterns that should have been skipped.

Commit:
Removes empty prometheus metrics

Micrometer eagerly initializes metrics for prometheus. Our actuate
metrics adapter was accidentally initializing metrics never written.

zeagord commented on zipkin-autoconfigure/metrics-prometheus/src/main/java/zipkin/autoconfigure/prometheus/ZipkinPrometheusMetricsAutoConfiguration.java:
```diff
@@ -18,6 +18,7 @@
 import io.micrometer.core.instrument.MeterRegistry;
```
We may not need this since we use the `PrometheusMeterRegistry`

codefromthecrypt replied:
HttpRequestDurationHandler is still referencing it.. maybe later, we will move that one to the zipkin-server, but I failed to get it working :P

Commit:
Reinstates old collector metrics naming convention

When we switched to micrometer's prometheus in 2.7.3, we accidentally
broke the naming convention used before.

Notably, in prometheus.yml, we had this:

```yaml
      - source_labels: [__name__]
        regex: '(?:gauge|counter)_zipkin_collector_(.*)_([^_]*)'
        replacement: '${2}'
        target_label: transport

```

As the default naming convention adds "_total", this causes dashboards
to end up with a transport literally named "_total".

Ex. counter_zipkin_collector_messages_http not counter_zipkin_collector_messages_http_total

zeagord committed:
Excludes health paths from prometheus configuration

codefromthecrypt commented:
reverted the naming convention as too many other things broke. going with the document approach:

For your prometheus, moving forward make sure your config has a _total on the end for the counters.
```diff
       # Received message count
       - source_labels: [__name__]
-        regex: '(?:gauge|counter)_zipkin_collector_(.*)_([^_]*)'
+        regex: '(?:gauge|counter)_zipkin_collector_(.*)_([^_]*)_total'
         replacement: '${2}'
         target_label: transport
       - source_labels: [__name__]
-        regex: '(?:gauge|counter)_zipkin_collector_(.*)_([^_]*)'
+        regex: '(?:gauge|counter)_zipkin_collector_(.*)_([^_]*)_total'
         replacement: 'zipkin_collector_${1}'
         target_label: __name__
```

Also, replace `http_requests_total` with `http_request_duration_seconds_count`
